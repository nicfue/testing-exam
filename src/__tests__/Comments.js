import React from 'react';
import { mount } from 'enzyme';
import Comments from '../components/Comments';
import * as api from '../api';


describe('testing comments', () => {
  const wrapper = mount(<Comments
    postId="1"
    currentPersona="fakeName"
  />)

  it('render comment component', () => {
    expect(wrapper.state()).toBeDefined()
  })

  it('render comment', () => {
    const comments = [
      {
        author: 'Zac',
        comment: 'test comment',
        date: '00000000',
        id: '1',
        postId: '1',
        currentPersona: ''
      }
    ];

    wrapper.setState({ comments });
    expect(wrapper.state().comments[0].comment).toContain("test comment");
  })

  const inst = wrapper.instance();

  it('call function removeComment from api', () => {
    api.removeComment = jest.fn();
    inst.removeComment()
    expect(api.removeComment).toHaveBeenCalled();
  })

  it('snapshot of comment', ()=> {
    expect(wrapper).toMatchSnapshot();
  })


});
