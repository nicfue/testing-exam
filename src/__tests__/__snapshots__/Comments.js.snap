// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing comments snapshot of comment 1`] = `
<Comments
  currentPersona="fakeName"
  postId="1"
>
  <div
    className="py-2"
  >
    <h2
      className="text-indigo-darker border-b mb-4"
    >
      Comments
    </h2>
    <SingleComment
      author="Zac"
      comment="comment test"
      currentPersona=""
      date="00000000"
      id="1"
      key="1"
      onClick={[Function]}
      postId="1"
    >
      <div
        className="py-2 border-b relative"
        key="1"
      >
        <p
          className="text-grey-darker mb-4"
        >
          comment test
        </p>
        <p
          className="text-grey-dark mb-4"
        >
          Posted by: 
          Zac
           @ 
          00000000
        </p>
      </div>
    </SingleComment>
    <CreateNewComment
      author="fakeName"
      postId="1"
      updateComments={[Function]}
    >
      <form
        className="container mx-auto flex flex-col p-6"
        onSubmit={[Function]}
      >
        <label
          className="mb-2"
          htmlFor="comment"
        >
          Comment
        </label>
        <textarea
          className="shadow rounded border mb-2 p-2"
          id="comment"
          name="comment"
          onChange={[Function]}
          value=""
        />
        <input
          className="bg-indigo-dark hover:bg-indigo-darker text-white font-bold py-2 px-4 rounded-full"
          type="submit"
          value="Comment"
        />
      </form>
    </CreateNewComment>
  </div>
</Comments>
`;
